src/vm/i386/umthunkstub.S

```diff
@@ -173,25 +173,20 @@ LOCAL_LABEL(UMThunkStub_CopyStackArgs):
 LOCAL_LABEL(InitCopyStack):
     push    ecx
     push    edx
-    mov     eax, 0  // Offset in edi
-    mov     ebx, 0  // Offset in esi
-LOCAL_LABEL(CopyStack):
     mov     edx, dword ptr [ebp - UMThunkStub_UMENTRYTHUNK_OFFSET]
     mov     edx, dword ptr [edx + UMEntryThunk__m_pUMThunkMarshInfo]
-    mov     ecx, dword ptr [edx + UMThunkMarshInfo__m_ecxArgOffset]
-    cmp     ecx, ebx
+    mov     ebx, [edx + UMThunkMarshInfo__m_cbActualArgSize]
```

Comment:
I'm not sure whether I understood your comment correctly, FunctionPtrTest and some Interop tests failed with changes below: 
```
LOCAL_LABEL(InitCopyStack):
    push    ecx
    push    edx
    mov     edx, dword ptr [ebp - UMThunkStub_UMENTRYTHUNK_OFFSET]
    mov     edx, dword ptr [edx + UMEntryThunk__m_pUMThunkMarshInfo]
    mov     ebx, [edx + UMThunkMarshInfo__m_cbActualArgSize]
    add     ebx, -4
LOCAL_LABEL(CopyStack):
    cmp     ebx, dword ptr [edx + UMThunkMarshInfo__m_ecxArgOffset]
    je      LOCAL_LABEL(IncreaseOffset)
    cmp     ebx, dword ptr [edx + UMThunkMarshInfo__m_edxArgOffset]
    je      LOCAL_LABEL(IncreaseOffset)
    add     eax, -4
    mov     ecx, dword ptr [esi + ebx]
    mov     dword ptr [edi + eax], ecx
LOCAL_LABEL(IncreaseOffset):
    add     ebx, -4
    jnc     LOCAL_LABEL(CopyStack)

    pop     edx
    pop     ecx
    jmp     LOCAL_LABEL(UMThunkStub_ArgumentsSetup)

```
Without these changes all tests passed, probably I missed something?